server/issue.go

```diff
@@ -63,85 +63,107 @@ func makePost(userID, channelID, message string) *model.Post {
+		return nil, http.StatusBadRequest, "missing post id"
 	}
 
 	originalPost, appErr := p.client.Post.GetPost(postID)
 	if appErr != nil || originalPost == nil {
-		return p.respondErrWithFeedback(authenticatedUserID, makePost(jiraBotID, channelID,
-			"post not found"), w, http.StatusNotFound)
+		return nil, http.StatusNotFound, "post not found"
 	}
+
 	if originalPost.UserId != jiraBotID {
-		return p.respondErrWithFeedback(authenticatedUserID, makePost(jiraBotID, channelID,
-			"user not authorized"), w, http.StatusUnauthorized)
+		return nil, http.StatusUnauthorized, "user not authorized"
 	}
 
 	if originalPost.ChannelId != channelID {
-		return p.respondErrWithFeedback(authenticatedUserID, makePost(jiraBotID, channelID,
-			"channel mismatch"), w, http.StatusBadRequest)
+		return nil, http.StatusBadRequest, "channel mismatch"
 	}
 
 	if requestData.UserId != "" && requestData.UserId != authenticatedUserID {
-		p.client.Log.Warn("share issue payload user mismatch",
+		p.client.Log.Warn(mismatchLog,
 			"header_user_id", authenticatedUserID,
 			"payload_user_id", requestData.UserId)
 	}
 
-	mattermostUserID := authenticatedUserID
-
 	val := requestData.Context["issue_key"]
 	issueKey, ok := val.(string)
 	if !ok {
-		return p.respondErrWithFeedback(mattermostUserID, makePost(jiraBotID, channelID,
-			"No issue key was found in context data"), w, http.StatusInternalServerError)
+		return nil, http.StatusInternalServerError, "No issue key was found in context data"
 	}
 	issueKey = strings.ToUpper(strings.TrimSpace(issueKey))
 	if !issueKeyRegexp.MatchString(issueKey) {
-		return p.respondErrWithFeedback(mattermostUserID, makePost(jiraBotID, channelID,
-			"invalid issue key"), w, http.StatusBadRequest)
+		return nil, http.StatusBadRequest, "invalid issue key"
 	}
 
 	val = requestData.Context["instance_id"]
 	instanceID, ok := val.(string)
 	if !ok {
-		return p.respondErrWithFeedback(mattermostUserID, makePost(jiraBotID, channelID,
-			"No instance id was found in context data"), w, http.StatusInternalServerError)
+		return nil, http.StatusInternalServerError, "No instance id was found in context data"
 	}
 
-	_, instance, connection, err := p.getClient(types.ID(instanceID), types.ID(mattermostUserID))
+	return &postActionContext{
+		authenticatedUserID: authenticatedUserID,
+		channelID:           channelID,
+		issueKey:            issueKey,
+		instanceID:          instanceID,
+		postID:              postID,
+	}, 0, ""
+}
+
+func (p *Plugin) httpShareIssuePublicly(w http.ResponseWriter, r *http.Request) (int, error) {
+	authenticatedUserID, requestData, status, err := decodePostActionRequest(r)
 	if err != nil {
-		return p.respondErrWithFeedback(mattermostUserID, makePost(jiraBotID, channelID,
+		return respondErr(w, status, err)
+	}
+
+	jiraBotID := p.getUserID()
+	ctx, status, msg := p.buildPostActionContext(authenticatedUserID, requestData, "share issue payload user mismatch")
+	if msg != "" {
```

Comment:
nit: for readability, can we rename this to errMsg? it's not clear what this msg is about